diff --git a/CMakeLists.txt b/CMakeLists.txt
index 935f623..52636ff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,7 +113,6 @@ add_library(FNA3D
 	src/FNA3D.c
 	src/FNA3D_Driver_D3D11.c
 	src/FNA3D_Driver_OpenGL.c
-	src/FNA3D_Driver_SDL.c
 	src/FNA3D_Image.c
 	src/FNA3D_PipelineCache.c
 	src/FNA3D_Tracing.c
@@ -124,7 +123,6 @@ add_library(mojoshader STATIC
 	MojoShader/mojoshader_common.c
 	MojoShader/mojoshader_d3d11.c
 	MojoShader/mojoshader_opengl.c
-	MojoShader/mojoshader_sdlgpu.c
 	MojoShader/profiles/mojoshader_profile_common.c
 	MojoShader/profiles/mojoshader_profile_glsl.c
 	MojoShader/profiles/mojoshader_profile_hlsl.c
@@ -141,7 +139,8 @@ endif()
 
 # Build flags
 if(NOT MSVC)
-	set_property(TARGET FNA3D PROPERTY COMPILE_FLAGS "-std=gnu99 -Wall -Wno-strict-aliasing -pedantic")
+	set_property(TARGET FNA3D PROPERTY COMPILE_FLAGS "-pthread -std=gnu99 -Wall -Wno-strict-aliasing -pedantic")
+	set_property(TARGET mojoshader PROPERTY COMPILE_FLAGS "-pthread")
 endif()
 if(BUILD_SHARED_LIBS)
 	set_property(TARGET mojoshader PROPERTY POSITION_INDEPENDENT_CODE ON)
