diff --git a/CMakeLists.txt b/CMakeLists.txt
index c2747d9..c4a1e08 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,9 +51,6 @@ endif()
 add_definitions(
 	-DFNA3D_DRIVER_OPENGL
 )
-if(BUILD_SDL3)
-	add_definitions(-DFNA3D_DRIVER_SDL)
-endif()
 add_definitions(
 	-DMOJOSHADER_NO_VERSION_INCLUDE
 	-DMOJOSHADER_USE_SDL_STDLIB
@@ -124,7 +121,6 @@ add_library(mojoshader STATIC
 	MojoShader/mojoshader_common.c
 	MojoShader/mojoshader_d3d11.c
 	MojoShader/mojoshader_opengl.c
-	MojoShader/mojoshader_sdlgpu.c
 	MojoShader/profiles/mojoshader_profile_common.c
 	MojoShader/profiles/mojoshader_profile_glsl.c
 	MojoShader/profiles/mojoshader_profile_hlsl.c
@@ -148,7 +144,8 @@ endif()
 
 # Build flags
 if(NOT MSVC)
-	set_property(TARGET FNA3D PROPERTY COMPILE_FLAGS "-std=gnu99 -Wall -Wno-strict-aliasing -pedantic")
+	set_property(TARGET FNA3D PROPERTY COMPILE_FLAGS "-pthread -std=gnu99 -Wall -Wno-strict-aliasing -pedantic")
+	set_property(TARGET mojoshader PROPERTY COMPILE_FLAGS "-pthread")
 endif()
 if(BUILD_SHARED_LIBS)
 	set_property(TARGET mojoshader PROPERTY POSITION_INDEPENDENT_CODE ON)
